devops/create_https_wss_endpoints_aws/main.py

```diff
@@ -0,0 +1,520 @@
+from collections import defaultdict
+
+import boto3
+import pprint
+from time import sleep
+import subprocess
+from dotenv import load_dotenv
+from timeit import default_timer as timer
+from datetime import timedelta
+
+
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+ap = argparse.ArgumentParser(description='parse the network type')
+ap.add_argument("-n", required=True, help="define network type")
+args = vars(ap.parse_args())
+
+current_work_path = os.path.dirname(os.path.realpath(__file__))
+network_config = current_work_path + '/' + args['n'] + '.json'
+
+
+
+# TO-DO: create the following dicts for each region
+# store target group arn, key: tg, value: arn of tg
+dict_tg_arn = dict()
+# store PREVIOUS instance id, key: tg, value: array of instance id
+dict_tg_instanceid = defaultdict(list)
+
+
+# store name of target group, key: tg_https, tg_wss, value: array of target group
+dict_tg_https_wss = defaultdict(list)
+
+
+# TO-DO: move to a helper class later
+def parse_network_config(param):
+    """ load the network configuration file, retrieve the value by its key """
+    with open(network_config, 'r') as f:
+        network_config_dict = json.load(f)
+
+    return network_config_dict[param]
+
+
+NUM_OF_REGIONS          = parse_network_config("num_of_regions")
+NUM_OF_SHARDS           = parse_network_config("num_of_shards")
+ARRAY_OF_REGIONS        = parse_network_config("regions")
+ARRAY_OF_VPC            = parse_network_config("region_vpc")
+ARRAY_OF_WS_ENDPOINTS   = parse_network_config("ws_endpoints")
+
+BASE_DOMAIN_NAME        = parse_network_config("domain_name")
+ID_DOMAIN_NAME          = BASE_DOMAIN_NAME.split('.')[0]
+array_domain_name       = []
+
+#### CREATE A COMPLETE PIPELINE ####
+dict_region_elb2arn     = defaultdict(list)
+dict_region_sslcerts    = defaultdict(list)
+dict_region_tgarn       = defaultdict(list)
+dict_region_ListenerArn = defaultdict(list)
+
+
+def create_endpoints_new_network():
+    """
+    COMPLETE PIPELINE
+    * 1/ - create SSL certificates (https, and wss) on each region
+    * 2/ - create Target Groups on each region
+    * 3/ - create ELB
+    * 4/ - create listener
+
+    * / - create entries on Route53
+    """
+
+    # 1/4 - request certificates
+    # test result: passed
+    request_ssl_certificates()
+
+    # 2/4 - create tg on each region
+    # test result: passed
+    create_target_group()
+
+    # 3/4 - create elb
+    # test result: passed
+    create_elb2()
+
+    # 4/ - create listener
+    # test result: passed
+    create_listener()
+
+    # 5/ - create rule the current listener
+    # test result: passed
+    create_rule()
+
+    # 6/ - register explorer instances into the target group
+    register_explorers()
+
+
+def register_explorers():
+    """
+    register explorer nodes into the corresponding target group
+        * register the same target into tg-https and tg-wss
+    depends on:
+        * dict_region_tgarn
+        *
+    """
+    for i in range(len(ARRAY_OF_REGIONS)):
```

Comment:
Range on array of regions directly , use enumerate if really need the index

Reply:
could do that too